packages/compass-sidebar/src/components/multiple-connections/sidebar.tsx

```diff
@@ -22,45 +32,139 @@ const sidebarStyles = css({
   height: '100%',
 });
 
-// TODO We will get rid of this placeholder when expose the necessary props outside
-// eslint-disable-next-line
-const noop_tmp = (() => {}) as any;
+type ConnectionErrorToastBodyProps = {
+  info: ConnectionInfo | null;
+  onReview: () => void;
+};
+
+const connectionErrorToastBodyStyles = css({
+  display: 'flex',
+  alignItems: 'start',
+  gap: spacing[2],
+});
 
+const connectionErrorToastActionMessageStyles = css({
+  marginTop: spacing[1],
+  flexGrow: 0,
+});
+
+function ConnectionErrorToastBody({
+  info,
+  onReview,
+}: ConnectionErrorToastBodyProps): React.ReactElement {
+  return (
+    <span className={connectionErrorToastBodyStyles}>
+      <span>
+        There was a problem connecting{' '}
+        {info ? `to ${getConnectionTitle(info)}` : ''}
+      </span>
+      {info && (
+        <Link
+          className={connectionErrorToastActionMessageStyles}
+          hideExternalIcon={true}
+          onClick={onReview}
+        >
+          REVIEW
+        </Link>
+      )}
+    </span>
+  );
+}
 // Having props here is useful as a placeholder and we will fix it with the first props.
 // eslint-disable-next-line
-export function MultipleConnectionSidebar({}: MultipleConnectionSidebarProps) {
+export function MultipleConnectionSidebar({
+  appName,
+}: MultipleConnectionSidebarProps) {
+  const { openToast, closeToast } = useToast('multiple-connection-status');
+  const cancelCurrentConnectionRef = useRef<(id: string) => Promise<void>>();
+
+  const [isExpanded, setIsExpanded] = useState(true);
+  const [isConnectionFormOpen, setIsConnectionFormOpen] = useState(false);
+
+  const onConnected = useCallback(() => {
+    closeToast('connection-status');
+  }, [closeToast]);
+
+  const onConnectionAttemptStarted = useCallback(
+    (info: ConnectionInfo) => {
+      const cancelAndCloseToast = () => {
+        void cancelCurrentConnectionRef.current?.(info.id);
+        closeToast('connection-status');
+      };
+
+      openToast('connection-status', {
+        title: `Connecting to ${getConnectionTitle(info)}`,
+        dismissible: true,
+        variant: 'progress',
+        actionElement: (
+          <Link hideExternalIcon={true} onClick={cancelAndCloseToast}>
+            CANCEL
+          </Link>
+        ),
+      });
+    },
+    [openToast, closeToast, cancelCurrentConnectionRef]
+  );
+
+  const onConnectionFailed = useCallback(
+    (info: ConnectionInfo | null, error: Error) => {
+      const reviewAndCloseToast = () => {
+        closeToast('connection-status');
+        setIsConnectionFormOpen(true);
+      };
+
+      openToast('connection-status', {
+        title: `${error.message}`,
+        description: (
+          <ConnectionErrorToastBody
+            info={info}
+            onReview={reviewAndCloseToast}
+          />
+        ),
+        variant: 'warning',
+        timeout: 5000,
```

Comment:
Very unexpected that error toast has a timeout, that's not a pattern we use anywhere else in the app

Reply:
Yeah Maurizio also was a bit surprised on this behaviour, but it's in the design. If the pattern is that error toasts are persistent, I think it's safe to remove the timeout and force the user to close it.